### PR TITLE
Add subject field to AlertPolicy's documentation block

### PR DESCRIPTION
**Pull request by joerayme**
Adds the subject field to the documentation block in AlertPolicy as per the [REST API documentation](https://cloud.google.com/monitoring/api/ref_v3/rest/v3/projects.alertPolicies#Documentation).

```release-note:enhancement
monitoring: Add `documentation.subject` field to `google_monitoring_alert_policy` resource
```

### CODE REVIEW

**Comment by joerayme** [comment]
I'm not sure why those tests would have failed: I haven't changed anything about Alloydb. I can't even find those tests in the repo and I can't access the error messages/debug logs to check what's going on either.

**Comment by c2thorn** [comment]
Don't worry about the alloydb tests, I can see they are clearly unrelated.
@joerayme have you seen https://github.com/GoogleCloudPlatform/magic-modules/pull/9172#issuecomment-1748686442 ?

**Comment by joerayme** [comment]
> have you seen [#9172 (comment)](https://github.com/GoogleCloudPlatform/magic-modules/pull/9172#issuecomment-1748686442) ?

Yes, just working on getting the corporate CLA signed which should hopefully be done shortly.

**Comment by c2thorn** [comment]
Great, let me know when that is finished and I will rerun our CI. In the meantime, I'll give a brief review

File: mmv1/third_party/terraform/services/monitoring/resource_monitoring_alert_policy_test.go [hunk 1]
```diff
@@ -322,6 +322,7 @@ resource "google_monitoring_alert_policy" "full" {
   documentation {
     content   = "test content"
     mime_type = "text/markdown"
+    subject   = "test subject"
```
Comment (c2thorn) [inline mmv1/third_party/terraform/services/monitoring/resource_monitoring_alert_policy_test.go:325]: I see there are no tests that try updating the values within `documentation`. We could use this opportunity and create another step in one of these tests similar to how the update test works: https://github.com/joerayme/magic-modules/blob/add-subject-to-alert-policy/mmv1/third_party/terraform/services/monitoring/resource_monitoring_alert_policy_test.go#L84

Alternatively, you could just add the `documentation` block to the update test with two separate values for each step.

Do you mind making one of these additions?

**Review by c2thorn — CHANGES_REQUESTED** [review]

File: mmv1/third_party/terraform/services/monitoring/resource_monitoring_alert_policy_test.go (hunk 1 above)
Comment (joerayme) [reply mmv1/third_party/terraform/services/monitoring/resource_monitoring_alert_policy_test.go:334]: OK, done. Sorry it took me a while. When adding the test I realised that the requirements of the documentation block weren't quite right so I've fixed those. I wasn't sure if there was a standard for validation tests. If there is, let me know and I can add one.

File: mmv1/products/monitoring/AlertPolicy.yaml [hunk 2]
```diff
@@ -966,20 +966,25 @@ properties:
     properties:
       - !ruby/object:Api::Type::String
         name: content
-        at_least_one_of:
-          - documentation.0.content
-          - documentation.0.mime_type
+        required: true
```
Comment (c2thorn) [inline mmv1/products/monitoring/AlertPolicy.yaml:969]: Even if this is a valid change, we cannot increasing requirements for an existing field is a breaking change. Breaking changes are only permitted in major releases (5.0.0 was recent). Can you revert the requirement changes and file [a new issue](https://github.com/hashicorp/terraform-provider-google/issues/new/choose) to track changing this in a future release?

**Review by c2thorn — CHANGES_REQUESTED** [review]

File: mmv1/products/monitoring/AlertPolicy.yaml (hunk 2 above)
Comment (c2thorn) [reply mmv1/products/monitoring/AlertPolicy.yaml:969]: Also, there are some merge conflicts that have popped up against the main branch. You will have to rebase/merge on the main branch

Comment (joerayme) [reply mmv1/products/monitoring/AlertPolicy.yaml:969]: There's a conflict because someone else got there first! #9360 Never mind, I'll close this now.

**Pull request closed without merging**